x/feegrant/keeper/keeper.go

```diff
@@ -51,54 +56,21 @@ func (k Keeper) GrantAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress,
 	store := ctx.KVStore(k.storeKey)
 	key := feegrant.FeeAllowanceKey(granter, grantee)
 
-	var oldExp *time.Time
-	existingGrant, err := k.getGrant(ctx, granter, grantee)
-
-	// If we didn't find any grant, we don't return any error.
-	// All other kinds of errors are returned.
-	if err != nil && !sdkerrors.IsOf(err, sdkerrors.ErrNotFound) {
+	exp, err := feeAllowance.ExpiresAt()
+	if err != nil {
```

Comment:
I thought the PR was only about moving the duplicate entry check from msg_server to keeper.

Can you explain what these changes are about?

Reply:
up

Reply:
Yes, the removed logic is for handling duplicate grant, which is unreachable if duplicate grant found.